configure.py

```diff
@@ -1990,8 +1991,8 @@ def query_seastar_flags(pc_file, link_static_cxx=False):
             obj = cc.replace('.cc', '.o')
             f.write('build {}: cxx.{} {}\n'.format(obj, mode, cc))
         f.write('build {}: cxxbridge_header\n'.format('$builddir/{}/gen/rust/cxx.h'.format(mode)))
-        librust = '$builddir/{}/rust-{}/librust_combined.a'.format(mode, mode)
-        f.write('build {}: rust_lib.{} | always\n'.format(librust, mode))
+        librust = '$builddir/{}/rust-{}/librust_combined'.format(mode, mode)
+        f.write('build {}.a: rust_lib.{} rust/Cargo.lock\n  depfile={}.d\n'.format(librust, mode, librust))
```

Comment:
After closer inspection, I can see that the depfile generated by `cargo` doesn't contain `Cargo.toml` files. Modifying those files should trigger recompilation, so I think we should include them on the list next to `Cargo.lock`.

Reply:
There's no easy way of adding all these `Cargo.toml` files to the script, but I updated the guidelines to require from developers to update the `Cargo.lock` file (by running `cargo update`) if they modify these `Cargo.toml` files, so even if we just depend on the `Cargo.lock` file, we should still notice the changes in `Cargo.toml`s and trigger recompilation

Reply:
OK, got it.

Reply:
> There's no easy way of adding all these Cargo.toml files to the script

What do you mean? Why?

Reply:
I meant an automatic way. Currently, we do require developers to add the `.rs` files that contain cxxbridge exports to the `configure.py` script, but the `Cargo.toml` files don't necessarily have to be tied to these files.
We could require adding all `Cargo.toml` files to the script manually, but we decided with @piodul that it's not necessary after all

Reply:
You could glob for them, if you wanted to do this automatically:
```
f.write('build {}.a: rust_lib.{} rust/Cargo.lock {}\n  depfile={}.d\n'
    .format(librust, mode, shlex.join(glob.glob("rust/**/Cargo.toml", recursive=True)), librust))
```

Reply:
I believe it would be possible, but I'm not completely sure a glob would be enough - when a new `Cargo.toml` file is added we would need to trigger not the build, but the `configure.py` script again.
The glob could also find some `Cargo.toml`s that we put there temporarily, but didn't actually use.